plugins/modules/ndo_l3out_node_group_policy.py

```diff
@@ -0,0 +1,366 @@
+            ),
+            aliases=["bfd_multi_hop"],
+        ),
+        target_dscp=dict(type="str", choices=list(TARGET_DSCP_MAP)),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["name"]],
+            ["state", "present", ["name"]],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    l3out = module.params.get("l3out")
+    name = module.params.get("name")
+    description = module.params.get("description")
+    node_routing_policy = module.params.get("node_routing_policy")
+    bfd = module.params.get("bfd")
+    target_dscp = TARGET_DSCP_MAP.get(module.params.get("target_dscp"))
+    state = module.params.get("state")
+
+    mso_template = MSOTemplate(mso, "l3out", template)
+    mso_template.validate_template("l3out")
+
+    l3out_object = mso_template.get_l3out_object(name=l3out, fail_module=True)
+    l3out_node_group = mso_template.get_l3out_node_group(name, l3out_object.details)
+
+    if name and l3out_node_group:
+        mso.existing = mso.previous = copy.deepcopy(l3out_node_group.details)  # Query a specific object
+    elif l3out_node_group:
+        mso.existing = l3out_node_group  # Query all objects
+
+    if state != "query":
+        node_group_policy_path = "/l3outTemplate/l3outs/{0}/nodeGroups/{1}".format(l3out_object.index, l3out_node_group.index if l3out_node_group else "-")
+
+    ops = []
+
+    if state == "present":
+        l3out_node_routing_policy_object = None
+        if node_routing_policy:
+            l3out_node_routing_policy_objects = mso.query_objs(
+                generate_api_endpoint(
+                    "templates/objects", **{"type": "l3OutNodePolGroup", "tenant-id": mso_template.template_summary.get("tenantId"), "include-common": "true"}
+                )
+            )
+            l3out_node_routing_policy_object = mso_template.get_object_by_key_value_pairs(
+                "L3Out Node Routing Policy", l3out_node_routing_policy_objects, [KVPair("name", node_routing_policy)], True
+            )
+
+        if mso.existing:
+            proposed_payload = copy.deepcopy(mso.existing)
+            if proposed_payload.get("bfdMultiHop", {}).get("key", {}).get("ref"):
+                proposed_payload["bfdMultiHop"]["key"].pop("ref", None)
+                mso.existing["bfdMultiHop"]["key"].pop("ref", None)
+
+            if description is not None and mso.existing.get("description") != description:
+                ops.append(dict(op="replace", path=node_group_policy_path + "/description", value=description))
+                proposed_payload["description"] = description
+
+            if l3out_node_routing_policy_object and mso.existing.get("nodeRoutingPolicyRef") != l3out_node_routing_policy_object.details.get("uuid"):
```

Comment:
use the function from https://github.com/CiscoDevNet/ansible-mso/pull/563/files#diff-f3e3720de5c754cfeb714aa1ed0f9d0a1a043f8c0005944d58565decefe1817bR31?